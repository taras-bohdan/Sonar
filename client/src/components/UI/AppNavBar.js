import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import AccountCircle from '@material-ui/icons/AccountCircle';
import { connect } from 'react-redux';


const styles = {
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -18,
    marginRight: 10,
  },
  grow: {
    flexGrow: 1,
  },
};

/**
 * Navigation bar
 * @param {Object} props - properties
 * @return {*} - component
 * @constructor
 */
function AppNavBar(props) {
  const { classes, authentication } = props;

  return (
    <div className={classes.root}>
      {authentication.loggedIn && (
        <AppBar position="static">
          <Toolbar variant="dense">
            <IconButton className={classes.menuButton} color="inherit" aria-label="Menu">
              <MenuIcon/>
            </IconButton>
            <Typography className={classes.grow} variant="h6" color="inherit">
              Sonar
            </Typography>
            <div>
              <IconButton
                aria-haspopup="true"
                color="inherit"
              >
                <AccountCircle/>
              </IconButton>
            </div>
          </Toolbar>
        </AppBar>
      )}
    </div>
  );
}

AppNavBar.propTypes = {
  classes: PropTypes.object.isRequired,
  authentication: PropTypes.object.isRequired,
};

/**
 * Map component's state to props to inject authentication state
 * @param {object} state - auth state
 * @return {{authentication: *}} - component's properties
 */
function mapStateToProps(state) {
  const { authentication } = state;
  return {
    authentication,
  };
}

export default withStyles(styles)(connect(mapStateToProps)(AppNavBar));
