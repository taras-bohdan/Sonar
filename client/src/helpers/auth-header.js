/**
 * Get authorization header from local storage
 * @returns {*} - auth token
 */
export function authHeader() {
  // return authorization header with jwt token
  const token = localStorage.getItem('userToken');

  if (token) {
    return { 'Authorization': 'Bearer ' + token };
  } else {
    return {};
  }
}
