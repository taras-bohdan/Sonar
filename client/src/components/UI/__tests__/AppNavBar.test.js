import React from 'react';
import { Provider } from 'react-redux';
import { shallow, render } from 'enzyme';

import AppNavBar from '../AppNavBar';
import { store } from '../../../helpers';

describe('<AppNavBar/>', () => {
  test('Renders without error', () => {
    shallow(<AppNavBar/>);
  });

  test('Renders successfully', () => {
    const wrapper = render(<Provider store={store}>
      <AppNavBar/>
    </Provider>);

    expect(wrapper).toMatchSnapshot();
  });
});
