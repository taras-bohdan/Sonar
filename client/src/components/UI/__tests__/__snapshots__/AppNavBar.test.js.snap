// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppNavBar/> Renders successfully 1`] = `
<div
  class="Connect-AppNavBar--root-1"
/>
`;
